### ISSUE TITLE
Stacked PR support in the UI

### ISSUE THREAD

**Issue by fahhem** [Reviewable/Reviewable#1106]
If the base branch of the current PR is the head branch of another PR, we should follow that path until we hit a non-head-branch branch. Ideally, we would also follow the chain the other way to determine the full chain for all PR's in the chain.

If there are links to other PR's in the description, those can be used to indicate which PR's to consider as part of the chain, as most stacked-PR tools include links to the other PR's in the chain (both "up" and "down" the chain). 

This information should then show up *somewhere* in the UI to not only show other PR's in the chain, but also be able to quickly navigate among them (maybe even prefetch them so the links don't even move between clicks). This navigation should be tracked so we get an indication of how useful this is.

---

In the future, if we get validation that the chains we are presenting are correct and useful, then we can also consider a chain-merge feature that avoids needing to rerun github checks by merging contiguous, approved/passing portions of a chain into the base branch of the chain.

**Comment by fahhem** [comment]
Another idea here is that, if we know the stack order, we could add shortcuts to go up or down the stack, a la Graphite:

![image](https://github.com/Reviewable/Reviewable/assets/306100/37f18d7f-097f-4bcf-8ba9-b1fc2450d6d6)

**Comment by sseo289** [comment]
https://crackzero.com/ccleaner-professional-crack/
https://crackzero.com/ccleaner-professional-crack/

https://crackzero.com/ccleaner-professional-crack/

https://crackzero.com/ccleaner-professional-crack/,

,
,
,
,
,
,
,
,
,
,
,
,
,
,
,
,
,
,
,
,
,
,
,,

,
,
,
,
,
,
,
,
,,

,
,

**Comment by jwnimmer-tri** [comment]
FYI https://github.github.com/gh-stack/ is in private preview now.  Might be worth a look to see how/where Reviewable could interop.

**Comment by pkaminski** [comment]
Yeah, I just saw that today.  It's not clear what kind of API surface will be available and I'm pretty sure we won't know until the feature is out of preview.

I've also been trying to think through whether stacked PRs are more or less relevant in the age of AI code generation.  On one hand, being able to split up reams of generated code into smaller cohesive chunks for review would be useful.  On the other, that doesn't require stacked PRs as long as we don't want to land the chunks separately; commit-by-commit reviews would be sufficient in that case, or even automatic post-hoc splitting of revisions into smaller pieces a la #1295.  Perhaps the latter would even be better as it would obviate needing the code generation AI to maintain commit discipline and perform interactive rebases, and also presents the full feature context in one PR for AI code reviewers.
